### PR TITLE
More configuration for prompt-able options

### PR DESCRIPTION
**Pull request by leightonthomas**
Adds default values, input value transformers, repeated asking of input and optional required-ness for options by using the Symfony OptionsResolver component

I was planning to use it a little more than I did which is why I used it in the first place but I think it works nicely for the initial configuration

In theory it shouldn't break any existing commands that use `addOption()` **and** `addPrompt()` because it checks if an option's already defined with that name

### CODE REVIEW

File: src/Transformer/FloatTransformer.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    public function transform($value)
+    {
+        // Not numeric and isn't blank (a value was actually provided)
+        if ( ! is_numeric($value) && ! $this->isBlank($value)) {
+            throw new LogicException("A valid float value is required.");
+        }
+
+        if ($value !== null) {
+            return (float) $value;
+        } else {
```
Comment (Brunty) [inline src/Transformer/FloatTransformer.php]: No need for the else here.

File: src/Transformer/IntegerTransformer.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    public function transform($value)
+    {
+        // Not numeric and isn't blank (a value was actually provided)
+        if ( ! is_numeric($value) && ! $this->isBlank($value)) {
+            throw new LogicException("A valid integer is required.");
+        }
+
+        if ($value !== null) {
+            return (int) $value;
+        } else {
```
Comment (Brunty) [inline src/Transformer/IntegerTransformer.php]: No need for the else here.

File: README.md [hunk 3]
```diff
@@ -111,9 +128,16 @@ When some options are not specified and the command is run with `--no-interactio
...
+## Option configuration
+
+The following configuration options are available per option:
 
-This has only been tested with string options set to `InputOption::VALUE_REQUIRED` - any other options are (currently) unsupported.
+| Configuration value | Description | Allowed values |
+|---------------------|-------------|----------------|
+| type | The type of value to be expected (input will be transformed) | `int`, `integer`, `float`, `bool`, `boolean`, `string` (written as strings)|
+| description | The description for the option/the text to be displayed when asking for the option | _anything_ |
+| required | Whether or not the option is required. If it's set to true, then a value MUST be entered and they will be prompted until one is provided. If set to false, a default value will be used if set, or `null` will be returned as their input. | `true` or `false` (as boolean)
```
Comment (Brunty) [inline README.md:139]: Just wondering why required is needed as an option, the point of the promptable stuff was to make options required anyway, trying to think of the use case for making something promptable but not required.

Would it be like a friendly "hey, you can supply this if you want?"

**Review by Brunty — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (leightonthomas) [reply README.md:139]: Yeah, I added it as more of "some sort of input is required for the code to function", for example if somebody forgot to provide an `age` but a value is necessary for the command to complete, you would set it to to be required, but you could also set it to false (which is the default) if it's an entirely optional thing code-wise.

File: src/Transformer/FloatTransformer.php (hunk 1 above)
Comment (leightonthomas) [reply src/Transformer/FloatTransformer.php]: Fixed in next commit

File: src/Transformer/IntegerTransformer.php (hunk 2 above)
Comment (leightonthomas) [reply src/Transformer/IntegerTransformer.php]: Fixed in next commit

**Review by Brunty — APPROVED** [review]

**Pull request merged**